### PR TITLE
Align the English doc with the Chinese doc

### PR DESCRIPTION
**Pull request by queenypingcap**
Move the MySQL compatibility page to the SQL folder.
Update the page with the latest information.

### CODE REVIEW

**Comment by queenypingcap** [comment]
@zimulala @shenli PTAL.

**Comment by queenypingcap** [comment]
@shenli @GregoryIan PTAL.

**Comment by queenypingcap** [comment]
@disksing PTAL.

**Comment by disksing** [comment]
_history-read.md_ LGTM.

**Comment by queenypingcap** [comment]
@iamxy @sunhao2017 PTAL

**Comment by queenypingcap** [comment]
@iamxy @sunhao2017 PTAL

File: sql/mysql-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+### Transaction
+
+TiDB implementsa an optimistic transaction model. Unlike MySQL, which uses row-level locking to avoid write conflict, in TiDB, the write conflict is checked only in the `commit` process during the execution of the statements like `Update`, `Insert`, `Delete`, and so on. 
+
+**Note:** On the business side, remember to check the returned results of `commit` because even there is no error in the execution, there might be errors in the `commit` process.
+
+
+### Load data
+
+When TiDB is in the execution of loading data, by default, a record with 20,000 rows of data is seen as a transaction for persistent storage. If a load data operation inserts more than 20,000 rows, it will be divided into multiple transactions to commit. If an error occurs in one transaction, this transaction in process will not be committed. However, transactions before that are committed successfully. In this case, a part of the load data operation is successfully inserted, and the rest of the data insertion fails. But aMySQL treats a load data operation as a transaction, one error leads to the failure of the entire load data operation.
```
Comment (iamxy) [inline sql/mysql-compatibility.md]: `aMySQL` typo?

**Comment by iamxy** [comment]
rest LGTM

**Comment by queenypingcap** [comment]
@GregoryIan PTAL the syncer user guide. Thanks!

**Comment by queenypingcap** [comment]
@winoros PTAL at the tidb-specific.md file. Thanks!
@shenli I believe we have other TiDB proprietary variables like tidb_snapshot. Can we also add them to this doc?

**Comment by queenypingcap** [comment]
@tiancaiamao PTAL at the privilege.md file. Thanks!

File: sql/privilege.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+	mysql -h 127.0.0.1 -P 4000 -u root
+	```
+ 
+## 2. Privilege-related operations
+ 
+### Grant privileges
+
+The `GRANT` statement grants privileges to the user accounts. 
+
+For example, use the following statement to grant the `xxx` user the privilege to read the `test` database.
```
Comment (tiancaiamao) [inline sql/privilege.md]: s/the `xxx` user/the user `xxx`

File: sql/privilege.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+ 
+```sql
+grant Select on test.* to 'xxx'@'%';
+```
+ 
+Use the following statement to grant the `xxx` user all privileges on all databases:
```
Comment (tiancaiamao) [inline sql/privilege.md]: ditto

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

File: sql/privilege.md (hunk 2 above)
Comment (queenypingcap) [inline sql/privilege.md]: Actually, we need to surround the name with "the" and the user. Same is true to command names like the xx command, etc.

**Comment by iamxy** [comment]
LGTM

**Comment by tiancaiamao** [comment]
privilege.md LGTM

**Pull request merged**